Issue 179 public header files script

Swift2568 opened the pull request:
Fixes #179

### Risk
This PR makes **no** API changes.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [ ] I have run the unit tests with this PR
- [ ] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).

#### Unit Tests
Ran script and verified that it only makes the desired changes.

#### Core Tests

Core version / branch / commit hash / module tested against: n/a (release script change)
HMI name / version / branch / commit hash / module tested against: n/a (release script change)

### Summary
Add a script publicscript.sh that will identify and fix header files that are not in the correct location

### Changelog
##### Enhancements
* Updates the release script to check that all files are properly public / private.

### Tasks Remaining:
n/a

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

Swift2568 commented:
[x] script identifies trouble files
 [] script moves files, and associated code files to correct location
 [] script update pathing in the project file

Swift2568 commented on scripts/projectfileheaderfix.sh:
```diff
@@ -0,0 +1,103 @@
+            fi
+        fi
+    fi
+done
```
Any file in public/ needs to be referenced in SmartDeviceLink.h.
Any file located in private/ needs to NOT be referenced in smartdevicelink.h

joeljfischer commented on scripts/projectfileheaderfix.sh:
```diff
@@ -0,0 +1,218 @@
+        
+            # Move the file to the correct destination
+            mv -f $file_found_location $destiny
+            echo "File "$header_filepath" moved to "$destiny"."
```
```suggestion
            echo "File "$header_filepath" moved to "$destiny""
```

joeljfischer commented on scripts/projectfileheaderfix.sh:
```diff
@@ -0,0 +1,218 @@
+            if [ ! -z "$code_file_found_location" ]; then
+                # Move associated code file.
+                mv -f $code_file_found_location $destiny
+                echo "File "$code_file" moved to "$destiny"."
```
```suggestion
                echo "File "$code_file" moved to "$destiny""
```

Swift2568 commented on scripts/projectfileheaderfix.sh:
```diff
@@ -0,0 +1,218 @@
+    echo "These files are public and were not found in "$project_header
+    for file_basename in $file_not_found_list
+    do
+        echo $file_basename
```
Applied the same change to the private files, as it might be useful to the user.

Swift2568 commented on scripts/projectfileheaderfix.sh:
```diff
@@ -0,0 +1,218 @@
+#!/bin/bash
```
That is an error with the move.  Instead of moving the file from where it was found to where it should be, I was moving the file from where the project file said it would be, to where it should be.  This makes sense for files changed in Xcode, but errors for files manually moved for testing.  I have now changed it to move the file from where the file is found, since this works in both scenarios.

joeljfischer commented on scripts/project_file_header_fix.sh:
```diff
@@ -0,0 +1,224 @@
+    echo "No private header files were found in "$project_header
+fi
+
+echo "Done checking public / private headers for correctness..."
+read user_input
```
```suggestion
echo "Done checking public / private headers for correctness"
```

This one wasn't supposed to have a `read user_input`

joeljfischer commented on scripts/project_file_header_fix.sh:
```diff
@@ -0,0 +1,224 @@
+            header_filepath="${header_file%%==*}"
+            header_type="${header_file##*==}"
+            code_file=$(sed -n 's/\.h/\.m/p' <<< "$header_filepath")
+            echo $header_filepath" and "$code_file" are marked "$header_type
```
The paths here have multiple `/` when they shouldn't:

```
SmartDeviceLink//public/SDLScreenManager.h and SmartDeviceLink//public/SDLScreenManager.m are marked private
```

joeljfischer commented on SmartDeviceLink/public/SmartDeviceLink.h:
```diff
@@ -488,6 +488,7 @@ FOUNDATION_EXPORT const unsigned char SmartDeviceLinkVersionString[];
 #pragma mark Touches
 #import "SDLPinchGesture.h"
 #import "SDLTouch.h"
+#import "SDLCapacityUnit.h"
```
This is not the location I mentioned in my review

joeljfischer commented on SmartDeviceLink/public/SmartDeviceLink.h:
```diff
@@ -497,6 +498,7 @@ FOUNDATION_EXPORT const unsigned char SmartDeviceLinkVersionString[];
 #import "SDLNotificationConstants.h"
 #import "SDLStreamingMediaManagerConstants.h"
 #import "SDLVersion.h"
+#import "SDLSystemInfo.h"
```
This is not the location I mentioned in my review

joeljfischer reviewed (changes requested):
Additional item:

1. I marked a public file as project and the script correctly moved the `.h` and `.m`. However, when it came to listing the `SmartDeviceLink.h` private files, this was the output:

```
These files are private and were found in SmartDeviceLink/public/SmartDeviceLink.h
Please remove them from the project header, then press enter to continue...
```

As you can see it should have listed the files to remove, but it did not.

Swift2568 commented on SmartDeviceLink/public/SmartDeviceLink.h:
```diff
@@ -488,17 +489,16 @@ FOUNDATION_EXPORT const unsigned char SmartDeviceLinkVersionString[];
 #import "SDLFunctionID.h"
 #import "SDLRPCFunctionNames.h"
 #import "SDLNotificationConstants.h"
+#import "SDLSystemInfo.h"
 #import "SDLStreamingMediaManagerConstants.h"
```
Alphabetical?  No problem.

Swift2568 replied:
for SDLManager.h
As I am trying to reproduce this, I do see the reference break.  Then Xcode automatically reloads the project file and the reference works again.

Swift2568 commented:
Issue has been reproduced and fixed.  The issue was a variable that was not being set properly.   I also fixed a bug where changes to file.m could accidentally change *file.m
